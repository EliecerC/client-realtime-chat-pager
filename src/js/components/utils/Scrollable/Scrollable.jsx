import React from 'react';
import PropTypes from 'prop-types';
import clx from 'classnames';
// styles
import './Scrollable.scss';

const Scrollable = React.forwardRef((props, ref) => {
  const { style, className, overflowX, overflowY, maxHeight, ...rest } = props;

  return (
    <div
      ref={ref}
      style={{
        maxHeight,
        overflowX,
        overflowY,
        ...style
      }}
      className={clx('scrollable-root', className)}
      {...rest}
    />
  );
});

Scrollable.displayName = 'Scrollable';

Scrollable.defaultProps = {
  className: '',
  overflowY: 'scroll',
  overflowX: 'hidden',
  maxHeight: '100%',
  style: {}
};

Scrollable.propTypes = {
  style: PropTypes.object,
  maxHeight: PropTypes.string,
  className: PropTypes.string,
  overflowX: PropTypes.string,
  overflowY: PropTypes.string
};

export default Scrollable;
