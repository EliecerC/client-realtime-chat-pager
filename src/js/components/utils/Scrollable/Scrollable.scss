.scrollable-root {
  width: 100%;
  position: relative;
  min-height: 200px;
  box-sizing: border-box;
  -moz-box-sizing: border-box;
  -webkit-box-sizing: border-box;
}