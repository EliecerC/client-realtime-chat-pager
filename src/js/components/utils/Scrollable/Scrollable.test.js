import React from 'react';
import renderer from 'react-test-renderer';
import Scrollable from './Scrollable.jsx';

/* eslint-disable no-undef */
it('renders without crashing', () => {
  // snapshot
  const scrollable = renderer.create(<Scrollable />).toJSON();
  expect(scrollable).toMatchSnapshot();
});
