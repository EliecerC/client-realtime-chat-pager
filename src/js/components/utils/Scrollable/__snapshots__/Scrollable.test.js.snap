// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="scrollable-root"
  style={
    Object {
      "maxHeight": "100%",
      "overflowX": "hidden",
      "overflowY": "scroll",
    }
  }
/>
`;
